pkg/controllers/management/clusteroperator/utils.go

```diff
@@ -166,13 +167,39 @@ func (e *OperatorController) CheckCrdReady(cluster *mgmtv3.Cluster, clusterType
 	return cluster, nil
 }
 
-func GenerateSAToken(restConfig *rest.Config, clusterName string) (string, error) {
-	clientSet, err := kubernetes.NewForConfig(restConfig)
+type TransportConfigOption func(*transport.Config)
+
+func WithDialHolder(holder *transport.DialHolder) TransportConfigOption {
+	return func(cfg *transport.Config) {
+		cfg.DialHolder = holder
+	}
+}
+
+func NewClientSetForConfig(config *rest.Config, opts ...TransportConfigOption) (*kubernetes.Clientset, error) {
+	transportConfig, err := config.TransportConfig()
 	if err != nil {
-		return "", fmt.Errorf("error creating clientset for cluster %s: %v", clusterName, err)
+		return nil, err
+	}
+	for _, opt := range opts {
+		opt(transportConfig)
+	}
+
+	rt, err := transport.New(transportConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	var httpClient *http.Client
+	if rt != http.DefaultTransport || config.Timeout > 0 {
```

Comment:
I failed to find the place where the [transport.New(transportConfig)](https://github.com/kubernetes/client-go/blob/v0.30.1/transport/transport.go#L36) function returns an `http.DefaultTransport`, could you provide more context here?

Reply:
Yeah, it's quite confusing.
 1. [`kubernetes.NewForConfig(restConfig)`](https://github.com/kubernetes/client-go/blob/v0.30.1/kubernetes/clientset.go#L468), which was used here, ensures the user agent is set, then calls:
 2. [`rest.HTTPClientFor(restConfig)`](https://github.com/kubernetes/client-go/blob/v0.30.1/rest/transport.go#L32), from which I copied these bits in order to be able to access `transportConfig`.

Reply:
Thanks for the links.